### PR TITLE
enable JSON format and confirm by pass

### PR DESCRIPTION
**Pull request by erichulburd**

### CODE REVIEW

**Comment by rigettizach** [comment]
I feel like I'm missing some of the design goals for this PR and I'd like to make sure I understand them before going off in the weeds on specific things.

I could imagine a `--confirm` option that means "When prompting for a yes/no question, always assume yes." But that's not quite what it means in this PR. Could you expand a bit about what `--confirm` means in the context of this PR?

I could imagine a `--format=json{-pretty}` as meaning "When producing output, make sure it's all machine-readable as JSON." But that's also not what it seems to mean in this PR - some commands still output non-JSON but also include some JSON. Could you clarify the goal so I know what to expect?

File: src/commands/cancel.ts [hunk 1]
```diff
@@ -31,24 +32,28 @@ export default class Cancel extends CommandWithCatch {
...
     const { flags } = this.parse(Cancel);
     const ids = parseValueOrValues(flags.id);
 
-    const resToCancel = (await GET.schedule({ ids })).filter(
-      r => r.status === 'ACTIVE',
-    );
-    if (resToCancel.length === 0) {
-      this.logErrorAndExit('reservation(s) found, but none that are active');
-    }
+    if (!flags.confirm) {
```
Comment (rigettizach) [inline src/commands/cancel.ts]: I think this error checking shouldn't be tied to the `--confirm` flag.

File: src/commands/reserve.ts [hunk 2]
```diff
@@ -66,16 +73,33 @@ export default class Reserve extends CommandWithCatch {
...
     }
 
-    const availreq = makeAvailabilityRequest(flags.start, flags.duration, flags.lattice);
+    if (flags.list) {
+      const availreq = makeAvailabilityRequest(flags.start, flags.duration, flags.lattice);
+      const availabilities = await GET.availability(availreq);
+      this.log(serializeAvailabilities(availabilities, flags.format as SerializeFormat));
+      return;
+    }
+    if (flags.confirm) {
```
Comment (rigettizach) [inline src/commands/reserve.ts:81]: I think it makes sense to have a separate flag for booking directly without having to select from a menu that's separate from the notion of `--confirm` overriding yes/no confirmation steps.

File: src/commands/reserve.ts [hunk 3]
```diff
@@ -66,16 +73,33 @@ export default class Reserve extends CommandWithCatch {
...
+      const reservations = await POST.reserve(resreq);
+      this.log(serializeReservations(reservations, { format: flags.format as SerializeFormat }));
+      return;
+    }
 
     let answer;
 
     do {
       const credits = await GET.credits();
       logCredits(credits);
```
Comment (rigettizach) [inline src/commands/reserve.ts:97]: This will output credit info in non-JSON format.

File: src/utils.ts [hunk 4]
```diff
@@ -278,9 +295,9 @@ export function getKey(keypath: string): string {
   return fs.readFileSync(pth, 'utf-8');
 }
 
-export async function confirmDeleteQMI(qmi: QMI): Promise<boolean> {
+export async function confirmDeleteQMI(qmi: QMI, format: SerializeFormat = 'tabular'): Promise<boolean> {
   console.log("Found 1 QMI for deletion:");
```
Comment (rigettizach) [inline src/utils.ts]: This will mix plain output with JSON output.

**Review by rigettizach — CHANGES_REQUESTED** [review]

File: src/commands/cancel.ts (hunk 1 above)
Comment (erichulburd) [reply src/commands/cancel.ts]: 👌

**Review by rigettizach — APPROVED** [review]

**Pull request merged**